lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -122,6 +125,46 @@ def render_pson(catalog)
       pson
     end
 
+    def clean_sensitive_parameters(catalog)
+      catalog['resources'].map! do |resource|
```

Comment:
Since you're using `map!`, the original catalog is modified, so the `clean_sensitive_parameters` methods doesn't need to return a catalog. It could just be renamed into `clean_sensitive_parameters!` and modify the catalog in place.

Reply:
Also done the same for `clean_nested_sensitive_parameters`